frontend/src/metabase/query_builder/components/QueryDownloadWidget/QueryDownloadWidget.jsx

```diff
@@ -114,30 +116,68 @@ const QueryDownloadWidget = ({
 
 const LOADER_SCALE_FACTOR = 0.9;
 
-const renderIcon = ({ icon, status, iconSize }) => {
+const renderIcon = ({ icon, status, iconSize, bordered }) => {
   if ([`idle`, `resolved`, `rejected`].includes(status)) {
     return (
       <Tooltip tooltip={t`Download full results`}>
-        <Icon
-          className="px1 py1"
-          data-testid="download-button"
-          title={t`Download this data`}
-          name={icon}
-          size={iconSize}
-        />
+        {bordered === true ? (
+          <BorderIcon
+            className="px1 py1"
+            data-testid="download-button"
+            title={t`Download this data`}
+            name={icon}
+            size={iconSize}
+          />
+        ) : (
+          <Icon
+            className="px1 py1"
+            data-testid="download-button"
+            title={t`Download this data`}
+            name={icon}
+            size={iconSize}
+          />
+        )}
       </Tooltip>
     );
   } else if (status === "pending") {
     return (
       <Tooltip tooltip={t`Downloading…`}>
-        <LoadingSpinner size={iconSize * LOADER_SCALE_FACTOR} />
+        {bordered === true ? (
+          <BorderLoadingSpinner size={iconSize * LOADER_SCALE_FACTOR} />
+        ) : (
+          <LoadingSpinner size={iconSize * LOADER_SCALE_FACTOR} />
+        )}
       </Tooltip>
     );
   } else {
     throw new Error(`Unknown download status: ${status}`);
   }
 };
 
+const BorderIcon = styled(Icon)`
+  padding: 0.6rem 1.25rem;
+  background: transparent;
+  border: 1px solid rgb(242, 236, 236);
+  border-radius: 6px;
+
+  &:hover {
+    background-color: #509ee357;
+    border-color: #c7dae3;
+  }
+`;
+
+const BorderLoadingSpinner = styled(LoadingSpinner)`
+  padding: 0.6rem 1.25rem;
+  background: transparent;
+  border: 1px solid rgb(242, 236, 236);
+  border-radius: 6px;
+
+  &:hover {
+    background-color: #509ee357;
+    border-color: #c7dae3;
+  }
+`;
+
```

Comment:
This can be moved to styled component file 
Also the css looks identical it can be made into a single styled component and Name can be Changed by inhereting from the Base component